src/test/java/org/gridsuite/sensitivityanalysis/server/NonEvacuatedEnergyTest.java

```diff
@@ -0,0 +1,873 @@
+        new IdentifiableAttributes("line3", IdentifiableType.LINE, null),
+        new IdentifiableAttributes("line4", IdentifiableType.LINE, null)
+    );
+    private static final List<IdentifiableAttributes> BRANCHES_2 = List.of(
+        new IdentifiableAttributes("line2", IdentifiableType.LINE, null),
+        new IdentifiableAttributes("newLine", IdentifiableType.LINE, null)
+    );
+
+    private static final String VARIANT_ID = "variant_1";
+
+    private static final int TIMEOUT = 1000;
+
+    private static final String ERROR_MESSAGE = "Error message test";
+
+    private static String INPUT;
+    private static String INPUT_WITH_TEMPORARY_LIMIT_NOT_FOUND;
+
+    @Autowired
+    private OutputDestination output;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private NetworkStoreService networkStoreService;
+
+    @MockBean
+    private ActionsService actionsService;
+
+    @MockBean
+    private FilterService filterService;
+
+    @MockBean
+    private ReportService reportService;
+
+    @MockBean
+    private UuidGeneratorService uuidGeneratorService;
+
+    @SpyBean
+    private NonEvacuatedEnergyWorkerService nonEvacuatedEnergyWorkerService;
+
+    @Value("${sensitivity-analysis.default-provider}")
+    String defaultSensitivityAnalysisProvider;
+
+    private final ObjectMapper mapper = RestTemplateConfig.objectMapper();
+
+    private List<NonEvacuatedEnergyStageDefinition> buildStagesDefinition() {
+        return List.of(NonEvacuatedEnergyStageDefinition.builder()
+                .generators(List.of(new EquipmentsContainer(GENERATORS_WIND_FILTER_UUID, "generators_wind")))
+                .energySource(EnergySource.WIND)
+                .pMaxPercents(List.of(100F, 70F)).build(),
+            NonEvacuatedEnergyStageDefinition.builder()
+                .generators(List.of(new EquipmentsContainer(GENERATORS_SOLAR_FILTER_UUID, "generators_solar")))
+                .energySource(EnergySource.SOLAR)
+                .pMaxPercents(List.of(70F, 50F)).build(),
+            NonEvacuatedEnergyStageDefinition.builder()
+                .generators(List.of(new EquipmentsContainer(GENERATORS_HYDRO_FILTER_UUID, "generators_hydro")))
+                .energySource(EnergySource.HYDRO)
+                .pMaxPercents(List.of(50F, 30F)).build());
+    }
+
+    private List<NonEvacuatedEnergyStagesSelection> buildStagesSelection() {
+        return List.of(NonEvacuatedEnergyStagesSelection.builder()
+                .name("EOL_100-PV_70-HYDRO_50")
+                .activated(true)
+                .stagesDefinitonIndex(List.of(0, 1, 2))
+                .pMaxPercentsIndex(List.of(0, 0, 0))
+                .build(),
+            NonEvacuatedEnergyStagesSelection.builder()
+                .name("EOL_70-PV_50-HYDRO_30")
+                .activated(true)
+                .stagesDefinitonIndex(List.of(0, 1, 2))
+                .pMaxPercentsIndex(List.of(1, 1, 1))
+                .build());
+    }
```

Comment:
I think here the model is not very clear. What about an API that would look like that ?
```suggestion
    private List<NonEvacuatedEnergyStages> buildStages() {
        NonEvacuatedEnergyGeneratorsSourceGroup windGenerators = NonEvacuatedEnergyGeneratorsSourceGroup.builder()
            .generators(List.of(new EquipmentsContainer(GENERATORS_WIND_FILTER_UUID, "generators_wind")))
            .energySource(EnergySource.WIND)
            .build();

        NonEvacuatedEnergyGeneratorsSourceGroup solarGenerators = NonEvacuatedEnergyGeneratorsSourceGroup.builder()
            .generators(List.of(new EquipmentsContainer(GENERATORS_SOLAR_FILTER_UUID, "generators_solar")))
            .energySource(EnergySource.SOLAR)
            .build();

        NonEvacuatedEnergyGeneratorsSourceGroup hydroGenerators = NonEvacuatedEnergyGeneratorsSourceGroup.builder()
            .generators(List.of(new EquipmentsContainer(GENERATORS_HYDRO_FILTER_UUID, "generators_hydro")))
            .energySource(EnergySource.HYDRO)
            .build();

        return List.of(
            NonEvacuatedEnergyStages.builder()
                .name("EOL_100-PV_70-HYDRO_50")
                .addGeneratorsSourceGroup(windGenerators, 1.)
                .addGeneratorsSourceGroup(solarGenerators, .7)
                .addGeneratorsSourceGroup(windGenerators, .5)
                .build(),
            NonEvacuatedEnergyStages.builder()
                .name("EOL_70-PV_50-HYDRO_30")
                .addGeneratorsSourceGroup(windGenerators, .7)
                .addGeneratorsSourceGroup(solarGenerators, .5)
                .addGeneratorsSourceGroup(windGenerators, .3)
                .build()
        );
    }
```